# Enables integer-only mode in FLAC when building with Vita SDK. Otherwise, we get an internal compiler error in FLAC__window_triangle:
#
# ../subprojects/flac/src/libFLAC/window.c: In function 'FLAC__window_triangle':
# ../subprojects/flac/src/libFLAC/window.c:197:1: error: unrecognizable insn:
#   197 | }
#       | ^
# (insn 191 190 192 21 (set (reg:V4SF 443)
#         (mult:V4SF (reg:V4SF 443)
#             (reg:V4SF 444))) "../subprojects/flac/src/libFLAC/window.c":189:43 -1
#      (nil))
# during RTL pass: vregs
# ../subprojects/flac/src/libFLAC/window.c:197:1: internal compiler error: in extract_insn, at recog.c:2294
# Please submit a full bug report,
# with preprocessed source if appropriate.
# See <https://gcc.gnu.org/bugs/> for instructions.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -198,6 +198,11 @@ if(CMAKE_BUILD_TYPE STREQUAL Debug OR CMAKE_BUILD_TYPE STREQUAL RelWithDebInfo)
     add_definitions(-DFLAC__OVERFLOW_DETECT)
 endif()
 
+check_symbol_exists(__vita__ "sys/config.h" VITA)
+if(VITA)
+    add_definitions(-DFLAC__INTEGER_ONLY_LIBRARY)
+endif()
+
 add_subdirectory("src")
 add_subdirectory("microbench")
 if(BUILD_DOCS)
